### DISCUSSION TITLE
Autonomous programming & reset of the ELOC board

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#78] [General]
@LIFsCode @EDsteve I wonder if you have a setup for autonomous programming & reset of the ELOC board? I've put together a circuit based on this [documentation](https://docs.espressif.com/projects/esptool/en/latest/esp32/advanced-topics/boot-mode-selection.html?highlight=boot#automatic-bootloader) that uses the DTR & RTS pins but the sticking point is the SD card. The card is not detected with the header pins are left connected to the circuit. 

Its come in useful for unit testing however.

![image](https://github.com/LIFsCode/ELOC-3.0/assets/40183643/0d657db5-9f25-4beb-8ac8-b24052bdfe6b)

**Comment by LIFsCode** [comment]
@OOHehir yes that has been [planned](https://github.com/LIFsCode/ELOC-3.0/issues/28) but I think it has been removed again, maybe because of space requirements on the PCB (@EDsteve )?

**Reply by OOHehir** [reply]
@LIFsCode Thanks. I might look at using a low voltage relay to isolate the SD pins when not flashing.

**Reply by EDsteve** [reply]
@LIFsCode @OOHehir 
That's correct. Due to space restrictions. The UART reset logic on-board forces the microphone to be too close to the pin headers which (we believe) was the reason why all 50 microphones got broken during manufacturing of the ELOC 3.3. Not sure if i want to take that risk again. JLCPCB is still investigation this issue though.
Another solution would be to expose the DTS and RST pins to the pinheaders and make a few programming-modules which have the UART reset logic (or even a FTDI chip) on the module and can be attached to the ELOC. Using that would make it possible to automatically flash the ELOC with the module and the microphone can stay put.
